services/QuillLMS/app/models/subscription.rb

```diff
@@ -80,15 +83,26 @@ class Subscription < ApplicationRecord
   SCHOOL_FIRST_PURCHASE_PRICE = SCHOOL_RENEWAL_PRICE
   TEACHER_PRICE = 8000
   ALL_PRICES = [TEACHER_PRICE, SCHOOL_RENEWAL_PRICE]
-  PAYMENT_METHODS = ['Invoice', 'Credit Card', 'Premium Credit']
+  PAYMENT_METHODS = [
+    INVOICE_PAYMENT_METHOD = 'Invoice',
+    CREDIT_CARD_PAYMENT_METHOD = 'Credit Card',
+    PREMIUM_CREDIT_PAYMENT_METHOD = 'Premium Credit'
+  ]
+
   ALL_TYPES = OFFICIAL_FREE_TYPES.dup.concat(OFFICIAL_PAID_TYPES)
 
+  validates :stripe_invoice_id, allow_blank: true, format: { with: /\Ain_[0-9a-zA-Z]*\z/ }
```

Comment:
Do we have documentation that this is a set format for `invoice.id`? Will it be loud if this fails (if they add an underscore or something).

Reply:
@dandrabik I couldn't find any documentation concerning the exact form, but all of the different models in the documentation followed this format.  I've added a validator: https://github.com/empirical-org/Empirical-Core/blob/da3ea2f9f97db42f1a9939fecb00a6553368666a/services/QuillLMS/app/models/validators/stripe_uid_validator.rb to give more informative error reporting when validation fails (e.g. subscription [creation](https://github.com/empirical-org/Empirical-Core/blob/da3ea2f9f97db42f1a9939fecb00a6553368666a/services/QuillLMS/app/services/stripe_integration/webhooks/subscription_creator.rb#L72)